Drivers/Inc/Minions.h

```diff
@@ -17,47 +28,41 @@
         PIN(BRAKELIGHT)  \
 
 typedef enum MINIONPIN_ENUM {
-    FOREACH_MinionPin(GENERATE_ENUM)
-    NUM_MINIONPINS,
-} MinionPin_t;
-
+    FOREACH_PIN(GENERATE_ENUM)
+    NUM_PINS,
+} pin_t;
 
-typedef enum{
-    MINION_ERR_NONE = 0,
-    MINION_ERR_WROTE_INPUT,
-} Minion_Error_t;
-
-
-typedef struct PinInfo{
+typedef struct {
     uint16_t pinMask;
     port_t port;
-    direction_t direction; //0 for input, 1 for output     
-} PinInfo_t;
-
+    direction_t direction;
+} pinInfo_t;
 
 /**
- * @brief Initializes input switches, output pins, and output mutex
+ * @brief Initializes digital I/O
  * 
  */
 void Minion_Init(void);
 
 /**
- * @brief Reads current state of specified input pin
+ * @brief Reads the status of a pin
  * 
- * @param pin specific pin to be read 
- * @return true
- * @return false *NOTE* If output pin is passed, will exit 
+ * @param pin 
+ * @return true is high
+ * @return false is low
  */
-bool Minion_Read_Pin(MinionPin_t pin, Minion_Error_t* err);
+bool Minion_Read(pin_t pin);
 
 /**
- * @brief Writes given status to a specified output pin. Locks writing to all output pins
+ * @brief Updates the status of a pin
  * 
- * @param pin specific pin to be written to 
- * @param status state of pin (0 or 1)
- * @return false if pin is not an output pin, true if it is
+ * @param pin 
+ * @param status 
+ * @return true is fail (wrote to an input)
```

Comment:
Any particular reason why this isn't the other way around?

Reply:
If we want to error check the function, it is cleaner to me to write `if (Minion_Read(...)) return error;` than `if (!Minion_Read(...)) return error;`

Just one less character to worry about I guess.

Reply:
We don't do that anywhere else, so it just feels a bit weird. But I guess it's fine

Reply:
I think returning false on fail makes a lot more sense. if (Minion_Read(...)) return error; looks like we return an error on a valid read